### DISCUSSION TITLE
[RTK Query]: Why is refetchOnMountOrArgChange named like this? Seems misleading

### DISCUSSION THREAD

**Discussion by domvo** [reduxjs/redux-toolkit#4040] [Q&A]
In query options you can pass an option called `refetchOnMountOrArgChange`. It defaults to false, you can set it to either true or give it a number.

There are several explanations in the docs. [One is here](https://redux-toolkit.js.org/rtk-query/usage/queries#query-hook-options):

> [refetchOnMountOrArgChange](https://redux-toolkit.js.org/rtk-query/api/createApi#refetchonmountorargchange) - Allows forcing the query to always refetch on mount (when true is provided). Allows forcing the query to refetch if enough time (in seconds) has passed since the last query for the same cache (when a number is provided). Defaults to false

[Another part](https://redux-toolkit.js.org/rtk-query/api/createApi#refetchonmountorargchange) in the docs says this:

> Defaults to false. This setting allows you to control whether if a cached result is already available RTK Query will only serve a cached result, or if it should refetch when set to true or if an adequate amount of time has passed since the last successful query result.

More often than not new team members are confused by this option and in particular why this option is called the way it's called. A more appropriate name would be something along the lines of `ignoreCachedValue`.

To the actual question: what is the motivation between the _OrArgChange_ portion of the name? Is this maybe related to old versions of rtk query in order to keep backwards compatibility? What does args refer to?

**Comment by phryneas** [comment]
Because it will trigger a refetch if the component is mounted and the cache entry is older than X or if the query argument changes (e.g. from `usePageQuery(1)` to `usePageQuery(2)`) and the cache entry for that query argument is older than X.

Also see https://github.com/rtk-incubator/rtk-query/issues/85 on naming.

**Reply by domvo** [reply]
> or if the query argument changes (e.g. from usePageQuery(1) to usePageQuery(2)) and the cache entry for that query argument is older than X.

Interesting, thank you for your quick response. Would you be so kind as to confirm the following thesis?

So it's specifically meant to force a refetch, if the argument changes and there is (still) a valid cache item for that argument in the store. If there is no data present in the store for that argument (because it has never been fetched or was invalidated), then the hook will trigger a fetch fetch on arg change, regardless of `refetchOnMountOrArgChange` being present.

**Reply by phryneas** [reply]
Yes, this option is always about refetching.
If no data is present, fetching will happen either way.
